### ISSUE TITLE
[Enhancement] Active Effects

### ISSUE THREAD

**Issue by Steelskinz** [Elemental-Re/crossblade#2]
First of. WOAW. I've been waiting so much time for a module like this one, being really fan of Divinity & JRPGs that have the same concept. Will it be possible to trigger it through an active effect later ?

**Comment by Elemental-Re** [comment]
I'm glad you like the module!

While I can envision of a couple of ways how a Crossblade event might be triggered via active effect, I'd like to understand how you imagine it should work. Would the active effect trigger an event and potentially fade in new layers for just the owner of the actor that receives the effect, or for everyone? Right now Crossblade works under the assumption that all clients basically hear the same song playing, the same as a sound in a playlist in core foundry, so triggering for just the owner would be a lot trickier to implement. Maybe you can give me an example of how you would like it to work in your game?

(I also added a seperate issue for the volume slider suggestion).

**Comment by Steelskinz** [comment]
Mh. This idea came to me because of an homebrew feature I gave to one of my players that can transform into an abyssal being. Like lycanthropy somehow, and Only when it would be its turn the theme would actually change to be a more dark theme. 
But It would be really nice to have a more " Speed up " theme when someone is bellow 25% by adding a active effect with Combat Utility Belt to it too, there's a lot of ideas and concepts than can flesh out from this.

**Comment by Elemental-Re** [comment]
Okay, I think I see the use case now. I'll take at the possibility of adding triggers on a per-client basis. At a minimum I'd need to finish implementing #1 first.

**Comment by Steelskinz** [comment]
I don't think you need it to be per client, mostly all will hear the music change. Like what's happening now when hostile,allies play. The only difference would be the condition of the trigger. Like : 

This music will play if a hostile creature plays : music.hostile.mp3

However, if hostile creature got activeEffect1  : music.activeeffect.1.mp3 will play instead.

Having a certain order of priority maybe?

**Comment by Elemental-Re** [comment]
Ah, I misunderstood. Okay, this should be a lot easier to do, once manual triggers are working.

**Comment by Steelskinz** [comment]
Yah I lack English grammar. Haha all fine! Love your work

**Comment by Elemental-Re** [comment]
With the release of [1.1.0](https://github.com/Elemental-Re/crossblade/releases/tag/1.1.0), I can now take a look at this issue. It should be possible to have this work now using the [Dynamic Active Effects](https://foundryvtt.com/packages/dae) module. Can I ask what system you are using?

**Comment by Steelskinz** [comment]
Hello and thanks for your work, I'm using d&d5e

**Comment by Elemental-Re** [comment]
I was working my way through an example but I just need to clarify one more thing.

So it's not that you want the effect to override the audible layers during every players turns, right? What you're actually looking for is just for the presence of the effect to override what would normally fade in during the effect target's turn. During other turns the layers would still operate as normal. Is that right?

**Comment by Steelskinz** [comment]
Yes you understood right!

**Comment by Elemental-Re** [comment]
Okay, I see now. So in order for this to work I still need to add another feature that's been part of my plans for some time which is combatant event overrides. Sorry for getting your hopes up this soon. I'm continuing to work on it though so hang in there.

**Comment by Steelskinz** [comment]
Hey it's fine, I'm already glad you're taking your time for that ! Don't overwork !
